app/lib/taric_synchronizer.rb

```diff
@@ -0,0 +1,137 @@
+class TaricSynchronizer
+  extend TariffSynchronizer
+
+  # 1 - does not raise an exception when record does not exist on TARIC DESTROY operation
+  #   - does not raise an exception when record does not exist on TARIC UPDATE operation
+  #   - creates new record when record does not exist on TARIC UPDATE operation
+  cattr_accessor :ignore_presence_errors
+  self.ignore_presence_errors = (ENV['TARIFF_IGNORE_PRESENCE_ERRORS'].to_i == 1)
+
+  cattr_accessor :username
+  self.username = ENV['TARIFF_SYNC_USERNAME']
+
+  cattr_accessor :password
+  self.password = ENV['TARIFF_SYNC_PASSWORD']
+
+  cattr_accessor :host
+  self.host = ENV['TARIFF_SYNC_HOST']
+
+  # Initial dump date + 1 day
+  cattr_accessor :initial_update_date
+  self.initial_update_date = Date.new(2012, 6, 6)
+
+  # TARIC query url template
+  cattr_accessor :taric_query_url_template
+  self.taric_query_url_template = '%{host}/taric/TARIC3%{date}'
+
+  # TARIC update url template
+  cattr_accessor :taric_update_url_template
+  self.taric_update_url_template = '%{host}/taric/%{filename}'
+
+  class << self
+    # Download pending updates for TARIC and CDS data
+    # Gets latest downloaded file present in (inbox/failbox/processed) and tries
+    # to download any further updates to current day.
+    def download
+      return Rails.logger.error 'Missing: Tariff sync enviroment variables: TARIFF_SYNC_USERNAME, TARIFF_SYNC_PASSWORD, TARIFF_SYNC_HOST and TARIFF_SYNC_EMAIL.' unless sync_variables_set?
+
+      TradeTariffBackend.with_redis_lock do
+        begin
+          TradeTariffBackend.patch_broken_taric_downloads? ? TariffSynchronizer::TaricUpdate.sync_patched : TariffSynchronizer::TaricUpdate.sync(initial_date: initial_update_date)
```

Comment:
We can kill the non-patched path, here.

They never fixed the api enough so that we can rely on the non-patched call _and_ they're migrating over to just dumping all of the downloader code in its current form for Taric code by the end of December

Reply:
This will be done in a separate ticket